numpy/core/tests/test_regression.py

```diff
@@ -2260,5 +2260,17 @@ def test_scalar_copy(self):
             item2 = copy.copy(item)
             assert_equal(item, item2)
 
+
+    def test_char_radd(self):
+        # GH issue 9620, used to reach gentype_add and raise TypeError
+        np_s = np.string_('abc')
+        np_u = np.unicode_('abc')
+        s = 'def'
+        u = u'def'
+        assert np_s.__radd__(np_s) is NotImplemented
+        assert np_u.__radd__(np_u) is NotImplemented
+        assert s + np_s == 'defabc'
```

Comment:
This fails on python 3. Try using the `b` prefix on the rhs.

Reply:
thanks, something went haywire with git, I am closing this and resubmitting as pull request #9629